src/WebJobs.Script.WebHost/ContainerManagement/LegionContainerInitializationHostService.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
+{
+    public class LegionContainerInitializationHostService : LinuxContainerInitializationHostService
+    {
+        private const string ContextFile = "Context.txt";
+
+        private readonly IEnvironment _environment;
+        private readonly ILogger _logger;
+
+        public LegionContainerInitializationHostService(IEnvironment environment, IInstanceManager instanceManager,
+            ILogger<LinuxContainerInitializationHostService> logger, StartupContextProvider startupContextProvider)
+            : base(environment, instanceManager, logger, startupContextProvider)
+        {
+            _environment = environment;
+            _logger = logger;
+        }
+
+        public override Task<string> GetStartContextOrNullAsync(CancellationToken cancellationToken)
```

Comment:
>GetStartContextOrNullAsync [](http://example.com/codeflow?start=37&length=26)

Might be cleaner to have

Task<bool> TryGetStartContext

And if it can't, just return false.

Reply:
Oh, right I forgot :| Maybe return a Tuple<bool, string> instead ?

Reply:
Made the changes. How does this look?

Reply:
Great, you just need to have 
`TryGetStartContext` now. 

It's implied to use bool here